packages/edit-site/src/components/list/added-by.js

```diff
@@ -20,6 +20,94 @@ import { _x } from '@wordpress/i18n';
 
 const TEMPLATE_POST_TYPE_NAMES = [ 'wp_template', 'wp_template_part' ];
 
+/**
+ *
+ * @typedef AddedByData
+ * @type {Object}
+ * @property {JSX.Element} icon         The icon to display.
+ * @property {string}      [imageUrl]   The optional image URL to display.
+ * @property {string}      [text]       The text to display.
+ * @property {boolean}     isCustomized Whether the template has been customized.
+ *
+ * @param    {Object}      template     The template object.
+ * @return {AddedByData} The added by object or null.
+ */
+export function useAddedBy( template ) {
```

Comment:
This is probably how I should have done it from the start 😄 

Thanks for improving it 🎉

Reply:
(passing objects is often a performance problem, as you can't optimize and have to re-render for any change in the object for the whole tree)

Reply:
Well, it can still be optimized by wrapping the hook inside an intermediate component, like how react-redux used to do:

```js
function AddedByText = ( { postType, postId } ) {
	const { record } = useEditedEntityRecord( postType, postId );
	const addedBy = useAddedBy( record );
	return addedBy.text;
}
```

A benefit of using the template object is that we don't have to depend on the `edit-site` store in this hook. It can be a _dumb_ hook (well, not really, as it still depends on the core data store), and a bit easier to unit test if needed.

I don't have a strong preference for this though, as I think either is fine and the performance impact doesn't matter much (until proven otherwise). Currently, the places that use this hook/component already have the `template` object in place, so I don't think it would matter in this situation.

Reply:
Yeah, I'd appreciate if we change this personally 👍 

I think the performance optimization are not negligible. I remember we had a similar issue in the past where the whole editor was re-rendering because we were passing the "template" around and any change to any block changes the template (blocks), so it rerenders all the components.

Reply:
Thanks for the update here.